src/librustc_session/config.rs

```diff
@@ -606,6 +606,7 @@ impl Default for Options {
             unstable_features: UnstableFeatures::Disallow,
             debug_assertions: true,
             actually_rustdoc: false,
+            allow_good_path_bugs: true,
```

Comment:
So with these defaults tools use path trimming and are slow by default.
I think it would be a safer default for tools to not use path trimming at all.

So the option needs to be about path printing specifically rather than about good path bugs in general, and it also needs to be a three variant enum rather than a boolean (https://github.com/rust-lang/rust/pull/73996#issuecomment-679355482).
```rust
enum TrimmedDefPaths {
    Never, // [default] `try_print_trimmed_def_path` never prints a trimmed path and never calls the expensive query
    Always, // `try_print_trimmed_def_path` calls the expensive query, the query doesn't call `delay_good_path_bug`
    GoodPath, // `try_print_trimmed_def_path` calls the expensive query, the query calls `delay_good_path_bug`
}
```